doc/INSTALL.md

```diff
@@ -57,8 +58,8 @@ Clone lightning:
     
 For development or running tests, get additional dependencies:
 
-    sudo apt-get install -y valgrind python3-pip libpq-dev
-    sudo pip3 install -r requirements.txt --use-feature=in-tree-build
+    sudo apt-get install -y valgrind python3-pip libpq-dev shellcheck cppcheck libsecp256k1-dev
+    pip3 install --user -r requirements.txt
```

Comment:
In version 21 of pip3 the feature `--use-feature` is still required, so I think we need to keep it until we are sure that there is no pip3 version 21 around
```suggestion
    pip3 install --user -r requirements.txt --use-feature=in-tree-build
```

Reply:
Hmm I am using version 20.3.4 and it simply breaks with `no such option: --user-feature`. Are we sure which versions require this?

Reply:
Ah ok your pip version is an old pip3 version https://github.com/pypa/pip/tags

Anyway, I think we need to document both in this case, and not just remove stuff because another user will put this change of a different env.

Reply:
I upgraded to 21.3.1 and it shows
```
DEPRECATION: In-tree builds are now the default. pip 22.1 will enforce this behaviour change. A possible replacement is to remove the --use-feature=in-tree-build flag.
```
So 20 breaks with `in-tree-build` and 21 already defaults to it. Which versions need this option to run?

Reply:
I suggest to documenting Before 21 and 21 because version 22 is the newest and we can stay safe with the warning.

Reply:
Ok, so looking into this more. Installing the contrib deps actually does not work with pip 20.3.4, because the `in-tree-build` option doesn't exist. Building it without that option causes the error `LookupError: setuptools-scm was unable to detect version for '/'.`. This is because without in-tree-builds the pyln-* dirs are copied to temp dirs before building, which makes them unable to detect version. I don't know why it worked for me initially before. Installation instructions need to upgrade pip first before installing requirements, which then will use in-tree-build by default. I've updated the instructions accordingly.